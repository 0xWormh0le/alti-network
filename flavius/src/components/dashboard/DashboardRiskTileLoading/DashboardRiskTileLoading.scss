@import 'styles/_shared.scss';

.DashboardRiskTileLoading {
  &__title {
    margin-bottom: 19px;
  }

  &__top {
    position: relative;
    display: flex;
    align-items: center;
  }

  &__avatar {
    width: 40px;
    height: 40px;
  }

  &__desc {
    margin-left: $BASE_SPACING;
    width: calc(100% - 40px - #{$BASE_SPACING});
    height: 43px;
    border-radius: 4px;
    overflow: hidden;
  }

  &__button {
    margin-top: $BASE_SPACING;
    width: 100%;
    height: 68px;
    border-radius: 4px;
    overflow: hidden;
  }
}
