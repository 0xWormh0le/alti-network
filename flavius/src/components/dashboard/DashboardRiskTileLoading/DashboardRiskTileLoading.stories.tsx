import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DashboardRiskTileLoading from './DashboardRiskTileLoading'

export default {
  title: 'Dashboard/DashboardRiskTileLoading',
  component: DashboardRiskTileLoading
} as Meta

const Template: Story<any> = (args) => <DashboardRiskTileLoading {...args} />

export const Default = Template.bind({})
Default.args = {}
