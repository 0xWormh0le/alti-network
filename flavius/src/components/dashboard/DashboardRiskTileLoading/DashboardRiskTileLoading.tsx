import React from 'react'
import ContentLoader from 'react-content-loader'

import './DashboardRiskTileLoading.scss'

export const DashboardRiskTileLoading = () => (
  <div className='DashboardRiskTile DashboardRiskTileLoading'>
    <ContentLoader
      backgroundColor='#eeeeee'
      foregroundColor='#fdfdfd'
      height={50}
      className='DashboardRiskTileLoading__title'
      uniqueKey='DashboardRiskTileLoading__title'>
      <rect x={0} y={0} width='100%' height={50} />
    </ContentLoader>
    <div className='DashboardRiskTileLoading__top'>
      <ContentLoader
        backgroundColor='#eeeeee'
        foregroundColor='#fdfdfd'
        height={30}
        width={30}
        className='DashboardRiskTileLoading__avatar'
        uniqueKey='DashboardRiskTileLoading__avatar'>
        <circle cx={20} cy={20} r={20} />
      </ContentLoader>
      <ContentLoader
        backgroundColor='#eeeeee'
        foregroundColor='#fdfdfd'
        height={30}
        className='DashboardRiskTileLoading__desc'
        uniqueKey='DashboardRiskTileLoading__desc'>
        <rect x={0} y={0} width='100%' height={40} />
      </ContentLoader>
    </div>
    <ContentLoader
      backgroundColor='#eeeeee'
      foregroundColor='#fdfdfd'
      height={67}
      className='DashboardRiskTileLoading__button'
      uniqueKey='DashboardRiskTileLoading__button'>
      <rect x={0} y={0} width='100%' height={67} />
    </ContentLoader>
  </div>
)

export default DashboardRiskTileLoading
