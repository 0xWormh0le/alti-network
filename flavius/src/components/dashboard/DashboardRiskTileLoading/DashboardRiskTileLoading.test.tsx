import React from 'react'
import { render } from '@testing-library/react'

import { DashboardRiskTileLoading } from './DashboardRiskTileLoading'

describe('DashboardRiskTileLoading', () => {
  it('renders correctly', () => {
    const { container } = render(<DashboardRiskTileLoading />)
    expect(container).toMatchSnapshot()
  })
})
