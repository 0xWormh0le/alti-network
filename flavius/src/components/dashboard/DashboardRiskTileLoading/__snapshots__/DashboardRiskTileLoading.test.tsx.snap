// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRiskTileLoading renders correctly 1`] = `
<div>
  <div
    class="DashboardRiskTile DashboardRiskTileLoading"
  >
    <svg
      aria-labelledby="DashboardRiskTileLoading__title-aria"
      class="DashboardRiskTileLoading__title"
      height="50"
      role="img"
    >
      <title
        id="DashboardRiskTileLoading__title-aria"
      >
        Loading...
      </title>
      <rect
        clip-path="url(#DashboardRiskTileLoading__title-diff)"
        height="100%"
        role="presentation"
        style="fill: url(#DashboardRiskTileLoading__title-animated-diff);"
        width="100%"
        x="0"
        y="0"
      />
      <defs
        role="presentation"
      >
        <clippath
          id="DashboardRiskTileLoading__title-diff"
        >
          <rect
            height="50"
            width="100%"
            x="0"
            y="0"
          />
        </clippath>
        <lineargradient
          id="DashboardRiskTileLoading__title-animated-diff"
        >
          <stop
            offset="0%"
            stop-color="#eeeeee"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1.2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-2; -2; 1"
            />
          </stop>
          <stop
            offset="50%"
            stop-color="#fdfdfd"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1.2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-1; -1; 2"
            />
          </stop>
          <stop
            offset="100%"
            stop-color="#eeeeee"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1.2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="0; 0; 3"
            />
          </stop>
        </lineargradient>
      </defs>
    </svg>
    <div
      class="DashboardRiskTileLoading__top"
    >
      <svg
        aria-labelledby="DashboardRiskTileLoading__avatar-aria"
        class="DashboardRiskTileLoading__avatar"
        height="30"
        role="img"
        width="30"
      >
        <title
          id="DashboardRiskTileLoading__avatar-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#DashboardRiskTileLoading__avatar-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#DashboardRiskTileLoading__avatar-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs
          role="presentation"
        >
          <clippath
            id="DashboardRiskTileLoading__avatar-diff"
          >
            <circle
              cx="20"
              cy="20"
              r="20"
            />
          </clippath>
          <lineargradient
            id="DashboardRiskTileLoading__avatar-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#eeeeee"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#fdfdfd"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#eeeeee"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
      <svg
        aria-labelledby="DashboardRiskTileLoading__desc-aria"
        class="DashboardRiskTileLoading__desc"
        height="30"
        role="img"
      >
        <title
          id="DashboardRiskTileLoading__desc-aria"
        >
          Loading...
        </title>
        <rect
          clip-path="url(#DashboardRiskTileLoading__desc-diff)"
          height="100%"
          role="presentation"
          style="fill: url(#DashboardRiskTileLoading__desc-animated-diff);"
          width="100%"
          x="0"
          y="0"
        />
        <defs
          role="presentation"
        >
          <clippath
            id="DashboardRiskTileLoading__desc-diff"
          >
            <rect
              height="40"
              width="100%"
              x="0"
              y="0"
            />
          </clippath>
          <lineargradient
            id="DashboardRiskTileLoading__desc-animated-diff"
          >
            <stop
              offset="0%"
              stop-color="#eeeeee"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-2; -2; 1"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#fdfdfd"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="-1; -1; 2"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#eeeeee"
              stop-opacity="1"
            >
              <animate
                attributeName="offset"
                dur="1.2s"
                keyTimes="0; 0.25; 1"
                repeatCount="indefinite"
                values="0; 0; 3"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
    <svg
      aria-labelledby="DashboardRiskTileLoading__button-aria"
      class="DashboardRiskTileLoading__button"
      height="67"
      role="img"
    >
      <title
        id="DashboardRiskTileLoading__button-aria"
      >
        Loading...
      </title>
      <rect
        clip-path="url(#DashboardRiskTileLoading__button-diff)"
        height="100%"
        role="presentation"
        style="fill: url(#DashboardRiskTileLoading__button-animated-diff);"
        width="100%"
        x="0"
        y="0"
      />
      <defs
        role="presentation"
      >
        <clippath
          id="DashboardRiskTileLoading__button-diff"
        >
          <rect
            height="67"
            width="100%"
            x="0"
            y="0"
          />
        </clippath>
        <lineargradient
          id="DashboardRiskTileLoading__button-animated-diff"
        >
          <stop
            offset="0%"
            stop-color="#eeeeee"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1.2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-2; -2; 1"
            />
          </stop>
          <stop
            offset="50%"
            stop-color="#fdfdfd"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1.2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-1; -1; 2"
            />
          </stop>
          <stop
            offset="100%"
            stop-color="#eeeeee"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1.2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="0; 0; 3"
            />
          </stop>
        </lineargradient>
      </defs>
    </svg>
  </div>
</div>
`;
